vmc/connector/clientconnector.go

```diff
@@ -82,15 +106,74 @@ func SecurityContextByRefreshToken(refreshToken string, cspURL string) (core.Sec
 		return nil, err
 	}
 
-	if res.StatusCode != 200 {
-		b, _ := io.ReadAll(res.Body)
-		return nil, fmt.Errorf("response from Cloud Service Provider contains status code %d : %s", res.StatusCode, string(b))
+	securityCtx, err := parseAuthnResponse(res)
+	if err != nil {
+		return nil, err
+	}
+	return securityCtx, nil
+}
+
+// newClientConnectorByClientID returns client connector to any VMC service by using OAuth authentication using clientId and secret.
+func newClientConnectorByClientID(clientID, clientSecret, serviceURL, cspURL string,
+	httpClient http.Client) (client.Connector, error) {
+
+	if len(serviceURL) <= 0 {
+		serviceURL = constants.DefaultVmcURL
+	}
+
+	if len(cspURL) <= 0 {
+		cspURL = constants.DefaultCspURL +
+			constants.CspOauthURLSuffix
+	} else {
+		cspURL = cspURL +
+			constants.CspOauthURLSuffix
+	}
+
+	securityCtx, err := securityContextByClientID(clientID, clientSecret, cspURL)
+	if err != nil {
+		return nil, err
+	}
+
+	connector := client.NewRestConnector(serviceURL, httpClient)
+	connector.SetSecurityContext(securityCtx)
+
+	return connector, nil
+}
+
+func securityContextByClientID(clientID string, clientSecret string, cspURL string) (core.SecurityContext, error) {
```

Comment:
Is there a reason not to use some standard packages to perform the OAuth2 protocol? like https://pkg.go.dev/golang.org/x/oauth2/clientcredentials ?

Reply:
No reason. I didn't know they existed. Reworked the mentioned fuction with the clientcredentials package.